tests/unit/package_managers/python/uv/test_main.py

```
@@ -0,0 +1,808 @@
+        ["uv", "lock", "--check", "--no-cache", "--no-python-downloads"],
+        params={"cwd": rooted_tmp_path.path},
+        suppress_errors=True,
+    )
+
+
+@mock.patch("hermeto.core.package_managers.python.uv.main.run_cmd")
+def test_validate_lockfile_fails_before_running_uv_when_lockfile_is_missing(
+    mock_run_cmd: mock.Mock, rooted_tmp_path: RootedPath
+) -> None:
+    """uv reports this as well, but names the temporary copy of the source instead."""
+    with pytest.raises(LockfileNotFound, match="uv.lock"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)